### ISSUE TITLE
private messages (PMs) in inbox : not sorted correctly, can't see recent ones, date is wrong, etc

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica#10003]
There's problems with seeing new messages in the "Messages" inbox:

When you see that there's a new message, you often (always?) have to reload before it will appear in the inbox. A sync is not enough. 

Conversations containing the most recent messages are not always at the top of the left-hand section of the Messages modal.

The date listed on the left-hand side is not always correct.

Here's an example of some of these problems from kAbi, child of Light (7adf6ada-3c05-4054-b5df-fa7d49d3b9eb):
"I just got a notification in the right corner about a new private message, but I can't find it. this is the second time this happened. since there's also the bug to sort messages wrongly, I'm not sure now whether I have a new messages or not. maybe I have one but it's sorted wrongly."
"right after posting the bug I found the message. it was from the second person from the list on the left. though on that list the date of the last time we did talk didn't update. refreshing the page. closing the tab and opening a new didn't chance any of that. the order of the messages is okay, and the person wrote me twice today (been 3h by now), but the list on the left still shows the person at the second slot and the date states 8 days ago."

**Comment by Alys** [comment]
Another example from AnneLine (16a107de-d519-43ce-8c30-8da5cbcbb237):
"Hi, Alys and all, is there a reason why all the PM messages are mixed up? For some reason my most recent message is at the bottom of each thread, but all the previous messages and any responses are at the top. I can live with it, but it makes it REALLY hard to follow messages! And I guess for now we just have to figure out who has sent the message from email and then search for that Habitican's message?"

**Comment by Alys** [comment]
Another problem: The most recent message in a conversation is not always shown in the left-hand side of the screen. The screenshot below was taken after reloading the website - i.e., a reload doesn't fix this problem.

![screenshot_2018-02-18_17-45-53](https://user-images.githubusercontent.com/1495809/36349477-adc22bba-14d3-11e8-80aa-a77e5ff5fa0d.png)

**Comment by Alys** [comment]
+1 on AnneLine's comment that I copied in above. I've only just seen a message she sent me a month ago - it must have been hidden amongst older messages from her the last time I looked.

**Comment by lemoness** [comment]
We're looking into this.
